qiskit-qir/src/qiskit_qir/visitor.py

```diff
@@ -0,0 +1,152 @@
+
+        labels = ", ".join([str(l) for l in qlabels + clabels])
+        if instruction.condition is None or skip_condition:
+            _log.debug(f"Visiting instruction '{instruction.name}' ({labels})")
+
+        if instruction.condition is not None and skip_condition is False:
+            _log.debug(f"Visiting condition for instruction '{instruction.name}' ({labels})")
+            conditions = [self._module.results[self._clbit_labels.get(bit)] for bit in instruction.condition[0]]
+
+            # Convert value into a bitstring of the same length as classical register
+            values = format(instruction.condition[1], f'0{len(results)}b')
+
+            # Add branches recursively for each bit in the bitstring
+            def __visit():
+                self.visit_instruction(instruction, qargs, cargs, skip_condition=True)
+
+            def _branch(conditions_values):
+                try:
+                    result, val = next(conditions_values)
+                    def __branch():
+                        self._builder.if_result(
+                            result=result,
+                            one=_branch(conditions_values) if val == "1" else None,
+                            zero=_branch(conditions_values) if val == "0" else None
+                        )
+                except StopIteration:
+                    return __visit
+                else:
+                    return __branch
+
+            _branch(zip(conditions, values))()
+        elif "measure" == instruction.name or "m" == instruction.name:
+            for qubit, result in zip(qubits, results):
+                self._builder.m(qubit, result)
+        elif "cx" == instruction.name:
+            self._builder.cx(*qubits)
+        elif "cz" == instruction.name:
+            self._builder.cz(*qubits)
+        elif "h" == instruction.name:
+            self._builder.h(*qubits)
+        elif "reset" == instruction.name:
+            self._builder.reset(qubits[0])
+        elif "rx" == instruction.name:
+            self._builder.rx(*instruction.params, *qubits)
+        elif "ry" == instruction.name:
+            self._builder.ry(*instruction.params, *qubits)
+        elif "rz" == instruction.name:
+            self._builder.rz(*instruction.params, *qubits)
+        elif "s" == instruction.name:
+            self._builder.s(*qubits)
+        elif "sdg" == instruction.name:
+            self._builder.s_adj(*qubits)
+        elif "t" == instruction.name:
+            self._builder.t(*qubits)
+        elif "tdg" == instruction.name:
+            self._builder.t_adj(*qubits)
+        elif "x" == instruction.name:
+            self._builder.x(*qubits)
+        elif "y" == instruction.name:
+            self._builder.y(*qubits)
+        elif "z" == instruction.name:
+            self._builder.z(*qubits)
+        elif "id" == instruction.name:
+            # See: https://github.com/qir-alliance/pyqir/issues/74
```

Comment:
https://github.com/qir-alliance/pyqir/pull/54 may also be relevant, as I'm not sure that there's a reserved name for an identity / wait instruction, such that it's up to each individual target to define an identity as part of their QIS if they want to support that call.